cyberdrop_dl/scraper/crawlers/nekohouse_crawler.py

```diff
@@ -0,0 +1,237 @@
+from __future__ import annotations
+
+import calendar
+import datetime
+from typing import TYPE_CHECKING, Tuple, Dict, Optional
+
+from aiolimiter import AsyncLimiter
+from bs4 import BeautifulSoup, Tag
+from yarl import URL
+
+from cyberdrop_dl.clients.errors import NoExtensionFailure
+from cyberdrop_dl.scraper.crawler import Crawler
+from cyberdrop_dl.utils.dataclasses.url_objects import ScrapeItem
+from cyberdrop_dl.utils.utilities import get_filename_and_ext, error_handling_wrapper, log
+
+if TYPE_CHECKING:
+    from cyberdrop_dl.managers.manager import Manager
+
+
+class NekohouseCrawler(Crawler):
+    def __init__(self, manager: Manager):
+        super().__init__(manager, "nekohouse", "Nekohouse")
+        self.primary_base_domain = URL("https://nekohouse.su")
+        self.services = ['fanbox', 'fantia', 'fantia_products', 'subscribestar', 'twitter']
+        self.request_limiter = AsyncLimiter(10, 1)
+        
+        self.post_selector = "article.post-card a"
+        self.post_content_selector = "div[class=scrape__files]"
+        self.file_downloads_selector = "a[class=scrape__attachment-link]"
+        self.post_images_selector = "div[class=fileThumb]"
+        self.post_videos_selector = "video[class=post__video] source"
+        self.post_timestamp_selector = "time[class=timestamp ]"
+        self.post_title_selector = "h1[class=scrape__title] span"
+        self.post_content_selector = "div[class=scrape__content]"
+        self.post_author_username_selector = "a[class=scrape__user-name]"
+
+    """~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"""
+
+    async def fetch(self, scrape_item: ScrapeItem) -> None:
+        """Determines where to send the scrape item based on the url"""
+        task_id = await self.scraping_progress.add_task(scrape_item.url)
+
+        if "thumbnails" in scrape_item.url.parts:
+            parts = [x for x in scrape_item.url.parts if x not in ("thumbnail", "/")]
+            link = URL(f"https://{scrape_item.url.host}/{'/'.join(parts)}")
+            scrape_item.url = link
+            await self.handle_direct_link(scrape_item)
+        elif "post" in scrape_item.url.parts:
+            await self.post(scrape_item)
+        elif any(x in scrape_item.url.parts for x in self.services):
+            await self.profile(scrape_item)
+        else:
+            await self.handle_direct_link(scrape_item)
+
+        await self.scraping_progress.remove_task(task_id)
+
+    @error_handling_wrapper
+    async def profile(self, scrape_item: ScrapeItem) -> None:
+        """Scrapes a profile"""
+        soup = await self.client.get_BS4(self.domain, scrape_item.url)
+        offset, maximum_offset = await self.get_offsets(scrape_item, soup)
+        service, user = await self.get_service_and_user(scrape_item)
+        user_str = await self.get_user_str_from_profile(soup)
```

Comment:
`user_str` is not used

Reply:
Just made a commit where we use user_str and also hopefully solve the issue with odd post references by passing pre-determined user data to the get_content functions